src/lib/components/Post.svelte

```diff
@@ -2,30 +2,37 @@
     import { PUBLIC_PREVIEW_HOST } from '$env/static/public';
 
     import * as Card from '$lib/shadcn/components/ui/card';
-    import * as Popover from '$lib/shadcn/components/ui/popover';
+    import * as DropdownMenu from '$lib/shadcn/components/ui/dropdown-menu';
+    import * as Dialog from '$lib/shadcn/components/ui/dialog';
+    import * as Select from '$lib/shadcn/components/ui/select';
+
+    import { Label } from '$lib/shadcn/components/ui/label';
+    import { Input } from '$lib/shadcn/components/ui/input';
     import { Badge } from '$lib/shadcn/components/ui/badge';
     import { Button } from '$lib/shadcn/components/ui/button';
     import { Skeleton } from '$lib/shadcn/components/ui/skeleton';
     import { Toggle } from '$lib/shadcn/components/ui/toggle';
 
     import ArrowDown from 'lucide-svelte/icons/arrow-down';
     import ArrowUp from 'lucide-svelte/icons/arrow-up';
-    import Share2 from 'lucide-svelte/icons/share-2';
     import Play from 'lucide-svelte/icons/play';
+    import EllipsisVertical from 'lucide-svelte/icons/ellipsis-vertical';
 
     import Score from './Score.svelte';
     import UserChannel from './UserChannel.svelte';
     import Player from './Player.svelte';
 
     import type { PostData } from '$lib/types';
     import { createEventDispatcher } from 'svelte';
+    import { toast } from 'svelte-sonner';
 
     export let post: PostData | undefined = undefined;
     export let playing_video: boolean = false;
     const dispatch = createEventDispatcher();
 
     let upvote_pressed = false;
     let downvote_pressed = false;
+    let open = false;
```

Comment:
Might want a more descriptive name for this variable like `report_dialog_open` or similar